File: veloProweb-frontend/src/app/models/DTO/sale-request-dto.ts
import { PaymentMethod } from "../enum/payment-method";
import { SaleDetailDTO } from "./sale-detail-dto";

export interface SaleRequestDTO {
    id: number;
    date: string;
    idCustomer: number | null;
    paymentMethod: PaymentMethod;
    tax: number;
    total: number;
    discount: number;
    numberDocument: number;
    comment: string;
    detailList: SaleDetailDTO[];
}

File: veloProWeb/src/main/java/com/veloProWeb/model/dto/sale/SaleRequestDTO.java
package com.veloProWeb.model.dto.sale;

import com.veloProWeb.model.Enum.PaymentMethod;
import lombok.Data;

import java.time.LocalDate;
import java.util.List;

@Data
public class SaleRequestDTO {
    private Long id;
    private LocalDate date;
    private Long idCustomer;
    private PaymentMethod paymentMethod;
    private int tax;
    private int total;
    private int discount;
    private int numberDocument;
    private String comment;
    private List<DetailSaleDTO> detailList;
}
